### PR TITLE
Improve GetFiles

### LINKED ISSUE #6: Avoid creating unnecessary processes
Hello 👋 

When deploying WinFIM.NET, I've noticed that we have lots of event log audit entries around the creation of processes (`cmd.exe`, `netstat.exe`, `qprocess.exe`). Would you be happy to receive PRs to avoid process creation and to use the underlying APIs where appropriate?

I can add some detail in the individual PRs.

### PR DESCRIPTION
**Pull request by slang25**
Partially fixes #6 

This change avoids spinning up a console host to enumerate files, whilst still including hidden files and not choking on inaccessible paths.

### CODE REVIEW

File: Project Source Code/WinFIM.NET Service/Controller.cs [hunk 1]
```diff
@@ -478,19 +443,39 @@ private static string[] GetExcludeList(string[] lines)
...
+            string[] directories = null;
+            try
+            {
+                files.AddRange(Directory.GetFiles(path, "*", SearchOption.TopDirectoryOnly));
+                directories = Directory.GetDirectories(path);
+            }
+            // Ignore inaccessible paths
+            catch (UnauthorizedAccessException) { }
+
+            foreach (var directory in directories)
```
Comment (mpowrie) [inline Project Source Code/WinFIM.NET Service/Controller.cs:467]: directories needs a null check
image

**Review by mpowrie — COMMENTED** [review]
Love the update! That also solves the problem of unicode directories such as c:\test\你好
If you can fix the null check in string[] directories, I think it looks good.

**Comment by slang25** [comment]
Thanks for catching that @mpowrie, updated 🙂

**Comment by mpowrie** [comment]
> Thanks for catching that @mpowrie, u> Thanks for catching that @mpowrie, updated 🙂

Looks good to me!
@redblueteam can you approve and merge the PR to master?

**Review by mpowrie — APPROVED** [review]

**Pull request merged**